test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/SortTests.cs

```diff
@@ -90,13 +91,82 @@ public async Task GivenMoreThanTenPatients_WhenSearchedWithSortParamWithHyphen_T
+        }*/
+
+        [Fact]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.SqlServer)]
+        public async Task GivenQueryWithDatetimeFilter_WhenSearchedWithSortParamOnDatetime_ThenResourcesAreReturnedInAscendingOrder()
+        {
+            var tag = Guid.NewGuid().ToString();
+
+            // save the timestamp prior to creating the resources
+            var now = DateTime.Now;
+            var time = now.AddSeconds(-10);
+
+            // create the resources which will have an timestamp bigger than the 'now' var
+            var patients = await CreatePatients(tag);
+
+            // Ask to get all patient with datetime filter
+            // Note: this might result in getting Patients that were created on other tests (concurrent tests),
+            // e.g. previous runs which were not cleaned yet, or concurrent tests.
+            // we overcome this issue by not looking for specific results, rather just make sure they are
+            // sorted.
+
+            // Format the time to fit yyyy-MM-ddTHH:mm:ss.fffffffzzz, and encode its special characters.
+            // sort and filter are based on same type (datetime)
+            string lastUpdated = HttpUtility.UrlEncode($"{time:o}");
+            await ExecuteAndValidateBundleSuperset($"Patient?_lastUpdated=gt{lastUpdated}&_sort=birthdate&_tag={tag}", false, patients.Cast<Resource>().ToArray());
+        }
+
+        [Fact]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.SqlServer)]
+        public async Task GivenQueryWithTagFilter_WhenSearchedWithSortParamOnDatetime_ThenResourcesAreReturnedInAscendingOrder()
+        {
+            var tag = Guid.NewGuid().ToString();
+
+            // create the resources which will have an timestamp bigger than the 'now' var
+            var patients = await CreatePatients(tag);
+
+            // Ask to get all patient with specific tag order by birthdate (timestamp)
+            // filter and sort are different based on different types
+            await ExecuteAndValidateBundle($"Patient?_tag={tag}&_sort=birthdate", false, patients.Cast<Resource>().ToArray());
+        }
+
+        [Fact]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.SqlServer)]
+        public async Task GivenQueryWithMultipleFilters_WhenSearchedWithSortParamOnDatetime_ThenResourcesAreReturnedInAscendingOrder()
+        {
+            var tag = Guid.NewGuid().ToString();
+
+            var patients = await CreatePatients(tag);
+
+            // Ask to get all patient with specific tag and family name order by birthdate (timestamp)
+            // filter and sort are different based on different types
+            var filteredFamilyName = "Williamas";
+            await ExecuteAndValidateBundle($"Patient?_tag={tag}&family={filteredFamilyName}&_sort=birthdate", false, patients.Reverse().Where(x => x.Name[0].Family == filteredFamilyName).OrderBy(x => x.BirthDate).Cast<Resource>().ToArray());
```

Comment:
I am wondering, in the current bundle if we filter by family name then it will return only one patient. as there is only one Patient with family name as "Williams. And if it returns just one patient, then order doesn't really matter. Am I missing something here?

Reply:
@eladiw

Reply:
@eladiw I fixed it.